pkg/controller/machinepool/nutanixactuator_test.go

```diff
@@ -0,0 +1,277 @@
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	machinev1 "github.com/openshift/api/machine/v1"
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hivev1nutanix "github.com/openshift/hive/apis/hive/v1/nutanix"
+)
+
+func getMasterMachineWithImage(t assert.TestingT) *machineapi.Machine {
+	masterMachine := testMachine("master0", "master")
+
+	uuid := "some-machine-image-uuid"
+	name := "nutanix-image-name"
+	nmpc := machinev1.NutanixMachineProviderConfig{
+		Image: machinev1.NutanixResourceIdentifier{
+			Type: machinev1.NutanixIdentifierName,
+			UUID: &uuid,
+			Name: &name,
+		},
+	}
+	providerSpecValue, err := json.Marshal(nmpc)
+	assert.NoError(t, err, "unexpected error creating NutanixActuator")
+
+	masterMachine.Spec.ProviderSpec.Value = &runtime.RawExtension{
+		Raw: providerSpecValue,
+	}
+
+	return masterMachine
+}
+
+func TestNewNutanixActuator(t *testing.T) {
+	actuator, err := NewNutanixActuator(getMasterMachineWithImage(t))
+	assert.NoError(t, err, "unexpected error creating NutanixActuator")
+	assert.NotNil(t, actuator, "expected a valid NutanixActuator instance")
+}
+
+func TestGenerateMachineSets(t *testing.T) {
+	tests := []struct {
+		name              string
+		clusterDeployment *hivev1.ClusterDeployment
+		pool              *hivev1.MachinePool
+		expectedErr       bool
+	}{
+		{
+			name: "ClusterDeployment with Nutanix platform and with proper metadata",
+			clusterDeployment: func() *hivev1.ClusterDeployment {
+				cd := testNutanixClusterDeployment()
+				cd.Spec.ClusterMetadata = &hivev1.ClusterMetadata{InfraID: "valid-infra"}
+				cd.Spec.Platform.Nutanix = &hivev1nutanix.Platform{
+					FailureDomains: []hivev1nutanix.FailureDomain{
+						{
+							Name: "Name",
+							PrismElement: hivev1nutanix.PrismElement{
+								UUID: "test-prism-element-uuid",
+								Endpoint: hivev1nutanix.PrismEndpoint{
+									Address: "valid-prism.example.com",
+									Port:    9440,
+								},
+								Name: "",
+							},
+						},
+					},
+				}
+				return cd
+			}(),
+			pool:        testNutanixPool(),
+			expectedErr: false,
+		},
+		{
+			name: "ClusterDeployment missing metadata",
+			clusterDeployment: &hivev1.ClusterDeployment{
+				Spec: hivev1.ClusterDeploymentSpec{
+					Platform: hivev1.Platform{
+						Nutanix: &hivev1nutanix.Platform{},
+					},
+				},
+			},
+			pool:        testNutanixPool(),
+			expectedErr: true,
+		},
+		{
+			name: "ClusterDeployment is not for Nutanix",
+			clusterDeployment: &hivev1.ClusterDeployment{
+				Spec: hivev1.ClusterDeploymentSpec{
+					ClusterMetadata: &hivev1.ClusterMetadata{InfraID: "test-infra"},
+				},
+			},
+			pool:        testNutanixPool(),
+			expectedErr: true,
+		},
+		{
+			name: "ClusterDeployment with multiple FailureDomains",
+			clusterDeployment: func() *hivev1.ClusterDeployment {
```

Comment:
In case you want to avoid this inline func thing for test data, we have some suites that use an option pattern, so e.g. this might instead look like:
```suggestion
			clusterDeployment: testNutanixClusterDeployment(WithFailureDomains(...)),
```

LMK if you want more info.